GetStream/stream-chat-android
# Fix MessageListViewModel initialization

closes #3379 

### 🎯 Goal

Fix `MessageListViewModel` initialization.

### 🛠 Implementation details

`ChatClient.watchChannelAsState` might return a null as the initial value or when the user is not initialized. We should subscribe to `ChannelData` updates and initialize the VM once `channelData != null` instead of checking the current value.

### 🎨 UI Changes

_Add relevant screenshots_

| Before | After |
| --- | --- |
| https://user-images.githubusercontent.com/17440581/164723861-d8d09c69-51dc-4414-98b6-8b93499cd4ef.mp4|https://user-images.githubusercontent.com/17440581/164723904-2b271759-8f82-4a8a-b25c-48db55f5d13a.mp4|

### 🧪 Testing
1. Go to channel -> channel details
2. Click on a member with whom you don't have 1-1 conversation
3. Click `message` option
4. You should be able to see a new, empty conversation

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist
